pkg/mads/v1/service_test.go

```diff
@@ -478,26 +478,15 @@ var _ = Describe("MADS http service", func() {
 			Expect(err).ToNot(HaveOccurred())
 			req.Header.Add("content-type", "application/json")
 
-			respChan := make(chan *http.Response, 1)
-
-			go func() {
-				resp, err := http.DefaultClient.Do(req)
-				Expect(err).ToNot(HaveOccurred())
-
-				respChan <- resp
-			}()
-
-			// given an updated mesh while the request is in progress
-			time.Sleep(5 * time.Millisecond)
-
-			err = createDataPlane(dp2)
+			start := time.Now()
+			resp, err := http.DefaultClient.Do(req)
+			// Ensure we're returning in less than 1 sec as this is the first request
+			Expect(time.Now()).To(BeTemporally("<", start.Add(time.Second)))
```

Comment:
I think that this is still flaky. If the request timed out after 1s, it can be some time after that when this check is run. I expect that we could use [httptrace](https://pkg.go.dev/net/http/httptrace) to get a more precise timing, but not sure if that's worth the effort. Maybe just add some wiggle room :)

Reply:
Actually maybe this test is unclear but this is the first request so it shouldn't wait 1s before returning. It should return straightaway.
I'll clarify the `It` to make the test clearer.

Reply:
Updated

Reply:
I saw the `fetch-timeout=1s` and assumed that this means the server should respond within that timeout, and the test looks like it is trying to verify that :shrug:

Reply:
I believe it's actually checking it returns immediately. Because originally it creates a 2nd dp 5ms after the request and the test asserts that we're getting only 1 dp. Which to me checks that the request is served before the 2nd dp is created. @austince can probably confirm if that's correct.
So I rewrote the test with less risk of race by just checking we're responding to this request in less than 1s